File: dclib/core/src/main/java/dclib/epf/systems/EntitySystem.java
package dclib.epf.systems;

import dclib.epf.Entity;
import dclib.epf.EntityManager;
import dclib.system.Updater;

public abstract class EntitySystem implements Updater {

	private final EntityManager entityManager;

	public EntitySystem(final EntityManager entityManager) {
		this.entityManager = entityManager;
	}

	@Override
	public final void update(final float delta) {
		for (Entity entity : entityManager.getAll()) {
			if (entity.isActive()) {
				update(delta, entity);
			}
		}
	}

	protected abstract void update(final float delta, final Entity entity);

}

File: dclib/core/src/main/java/dclib/epf/EntitySystem.kt
package dclib.epf

import dclib.system.Updater

abstract class EntitySystem(private val entityManager: EntityManager) : Updater {
    override fun update(delta: Float) {
        for (entity in entityManager.getAll()) {
            if (entity.isActive) {
                update(delta, entity)
            }
        }
    }

    protected abstract fun update(delta: Float, entity: Entity)
}